_code/InSituTree/R/runInSituTree.R

```diff
@@ -0,0 +1,158 @@
+    stop("Error: quantile_absolute_expression_difference_param must be a numeric value between 0 and 1.")
+  }
+
+  if (!is.numeric(quantile_percent_expression_difference_param) || quantile_percent_expression_difference_param < 0 || quantile_percent_expression_difference_param > 1) {
+    stop("Error: quantile_percent_expression_difference_param must be a numeric value between 0 and 1.")
+  }
+
+  out <- list()
+
+  # Make combined profile
+  prof <- collapseProfiles(cth = cth, full_profiles = full_profiles)
+
+  # annotate with InSituType
+  res <- supervisedSubcluster(reference_profiles = prof,
+                              x = x,
+                              neg = neg,
+                              cohort = cohort,
+                              excluded_genes = excluded_genes,
+                              quantile_absolute_expression_difference = quantile_absolute_expression_difference_param,
+                              quantile_percent_expression_difference = quantile_percent_expression_difference_param
+  )
+  out[[name_for_new_annotation]] <- list()
+  out[[name_for_new_annotation]][["result"]] <- res
+  out[[name_for_new_annotation]][["subclusterings"]]<- list()
+  out[[name_for_new_annotation]][["name"]] <- name_for_new_annotation
+
+  # annotation next level down
+
+  for(i in names(cth)){
+    if(i == "list" || i == name_for_new_annotation || i == ""){
+      next}
+    message(i)
+    celltypes_dropped <- !i %in% unique(res$clust) # check to make sure further subtyping is required.  This is where cell lineages that are not present are dropped out.
+    if(celltypes_dropped){
+      message(paste0("Warning: No cells found on which to perform further fitting.  Dropping celltypes: ", i))
+    }
+
+    if(length(cth[[i]]) > 1 & !celltypes_dropped){
+      message(paste0("Annotating  ", i, " cells..."))
+      selected_cells <- match( names(res$clust)[res$clust == i], rownames(x) )
+      sub_res <- runInSituTree(full_profiles = full_profiles
+                               , cth = cth[[i]]
+                               , x = x[selected_cells , ]
+                               , neg = neg[selected_cells]
+                               , name_for_new_annotation = i
+                               , cohort = cohort[selected_cells,]
+                               , excluded_genes = excluded_genes
+                               , quantile_absolute_expression_difference_param = quantile_absolute_expression_difference_param
+                               , quantile_percent_expression_difference_param = quantile_percent_expression_difference_param
+                               , return_summary_annotation = FALSE
+      )
+      out[[name_for_new_annotation]][["subclusterings"]] <- append(out[[name_for_new_annotation]][["subclusterings"]], sub_res)
+    }
+  }
+
+  # return the nested list of insitutype objects
+  if(return_summary_annotation){
+    summaryAnnotation <- summarizeInSituTree(out[[1]])
```

Comment:
This would include zero-count cells showed up in later subclustering, But cells with zero-count cell at root level would not be included in the final results. We should include them with `NA` value at `annotLevel_1`.

Reply:
Agreed. I've made this change.